wiki/BBCode/en.md

```diff
@@ -1,414 +1,294 @@
+
+![Quote reply button](/img/quotereply.png)
+
+### Code block
+
+```
+[code]
+text
+[/code]
+```
+
+The `[code]` tag is used to create *preformatted code blocks* (a.k.a. *preformatted text*). In the osu! forums, the `[code]` tag will format text in a monospace font inside of a semi-transparent gray box. Formatting text inside of a code block will tell the editor to treat the text between those tags literally, thus preventing the conversion of any tags or source code into something else.
+
+Within the osu! forums, code blocks are most often used to post source code for a [storyboard](/wiki/Storyboards), or in tutorials that require showing the syntax for tags, commands, or source code.
 
 ### Centre
 
-Centre align a paragraph; this is typically used for titles or headings.
+```
+[centre]text[/centre]
+```
 
-- Dedicated button: N/A
-- Notes:
-  - The tags **must** be spelt as `centre` not `center` (note the last two letters).
-- Syntax:
-  ```
-  [centre]
-  ...
-  [/centre]
-  ```
+The `[centre]` tag is used to align text to the centre of a post. This is most often used for stylistic effect in titles, headers, or poems. If placed inside of or surrounding a `[quote]` tag, the text inside the quote block will be centred, but the stylistic lines and such will not.
 
 ### URL
 
-Adds a link.
+```
+[url=LINK]text[/url]
+```
 
-You do not need to use this tag if you do not want to use link text. The osu!forums will automatically link pasted URLs inside the post.
+The `[url]` tag is used to turn regular text into clickable hyperlinks.
 
-- Dedicated button: ![URL button](img/url.png)
-- Notes:
-  - The `LINK` argument should not use quotation marks.
-- Syntax:
-  - Where `LINK` is the url.
-  - Where `TEXT` is the link text.
-  ```
-  [url=LINK]TEXT[/url]
-  ```
+*Note: the use of this tag is not necessary should one wish not to use custom hyperlink text, as the forum editor will automatically link any URLs that it detects.*
 
-### Profile
+To create hyperlinks with the `[url]` tag, users must specify two arguments: the linked text to be displayed, and the specific URL of the website to navigate to. The former must be specified between the start and end tags, and the latter must be specified as the `LINK` argument, without quotation marks (`"`). If no text is specified, the text will default to the name of the URL.
+
+Toolbar button: ![URL button](img/url.png)
 
-Links a user profile using the user's name.
+### Profile
 
-Usage of this button is **heavily discouraged**! The problem with using this tag is that user names can be changed once after obtaining an [osu!supporter tag](/wiki/osu!supporter). And once they do, the link will fail.
+```
+[profile]username[/profile]
```

Comment:
`[profile=userid]text[/profile]` works as well. in this case link does not break when the user in question changes their name, and whatever you type as `text` isn't displayed either - replaced by the username instead. you should add this as another usage of the profile tag